docs/arcaflow/workflows/expressions.md

```diff
@@ -224,14 +287,16 @@ The mathematical expression 2<sup>3</sup> is represented in the expression langu
 
 ### Equal To
 
-The `==` operator checks for equality between the left and right type. It returns true when the left and right match. The type must be the same for both operands, so the expression `1 == 1.0` would fail.
+The `==` operator checks for equality between the left and right type. It returns true when the left and right match. The type must be the same for both operands, so the expression `1 == 1.0` would produce a type error. You may use functions to convert types. For example, `intToFloat(1) == 1.0`. See [functions](#functions) for more type conversions.
+
 The operator currently supports the types `integer`, `float`, `string`, and `boolean`. If another type is required, please create an issue with the expected behavior of the operator with the needed type.
 
 For example, `2 == 2` results in `true`, and `"a" == "b"` results in `false`.
 
 ### Not Equal To
 
-The `!=` operator is the inverse of the [==](#equals ) operator. It returns true when the values do not match. The type must be the same for both operands, so the expression `1 != 1.0` would fail
+The `!=` operator is the inverse of the [==](#equals ) operator. It returns true when the values do not match. The type must be the same for both operands, so the expression `1 != 1.0` would produce a type error. You may use functions to convert types. For example, `intToFloat(1) != 1.0`. See [functions](#functions) for more type conversions.
```

Comment:
In some sense, `!=` and `==` are only inverses when applied to _numbers_ (and, by extension, to collections, such as strings, lists, maps, objects, etc.).  Since `NaN`'s are _not numbers_, expecting those operators to have the same relationships when they are applied to those values is perhaps not appropriate.

Reply:
> the behavior of the operators is usually described in terms of _types_ rather than discrete _values_

True, and that bothered me as I was composing my earlier reply (although, as you pointed out in your original comment, a `NaN` is not necessarily a discrete value -- there are several representations which prompt the "is-NaN" function to return true).

> the behavior of floating point operations on that value is not well behaved

Ah, but now you're getting judgey.  What constitutes "well behaved"?  We have definitions for the behavior of each of the mathematical operators when either of the operands is a NaN; likewise, we have definitions for the behavior of each of the relational operators when either of the operands is a NaN; so, why do you say that the conversion function is not well behaved if we know exactly what result it will produce?

> My point was that describing != as ! == isn't strictly accurate in all cases, and I wasn't necessarily suggesting a change ... but describing the two operations independently instead of documenting an inverse relationship somewhat mitigates the oddity.

❤️ 

> But, since the documentation does talk about NaN and Infinities in other contexts, a note here that NaN is "unordered" and is neither == nor != to any value (including itself) wouldn't entirely be out of place ... 😆

I mostly agree, but if we're going to have a note here, we should have a similar note for each of the other relational operators.

>[!Warning]
> But...Dave, could you check on your source which says that `NaN != NaN` yields `false`?  I believe that it should yield _`true`_ (and Go appears to back me up on this).  That is, `!(NaN == NaN) == (NaN != NaN)`, because no NaN is equal to any other NaN, and, therefore `!=` returns `true` when either operand is a `NaN`, and therefore, `!=` _is_ the inverse of `==` (not that I recommend describing it that way...).

And, I think that removes the apparent inconsistency that we're concerned about.

Reply:
Python also agrees that `nan != nan` yields `true`. Sorry: apparently I misinterpreted a convoluted phrase, and I investigated again more closely to the IEEE source. Essentially, for all floating point values including infinities, `x == x` yields true and `x != x` yields false while for NaN values, the reverse is true, `x == x` yields false and `x != x` yields true.